napari_btrack/track.py

```diff
@@ -31,23 +148,281 @@ def run_tracker(objects, config_file_path):
+            )
+            motion_model_dict[widget.name] = matrix
+        else:
+            if widget.name in motion_model_keys:
+                motion_model_dict[widget.name] = widget.value
+        # setup hypothesis model
+        if widget.name in hypothesis_model_keys:
+            hypothesis_model_dict[widget.name] = widget.value
+        if widget.name in all_hypotheses:  # hypotheses need special treatment
+            if getattr(container, widget.name).value:
+                hypotheses.append(widget.name)
+
+    # add some non-exposed default values to the motion model
+    mode = getattr(container, "mode").value
+    for default_name, default_value in zip(
+        ["measurements", "states", "dt", "prob_not_assign", "name"],
+        [3, 6, 1.0, 0.001, f"{mode}_motion"],
+    ):
+        motion_model_dict[default_name] = default_value
+
+    # add some non-exposed default value to the hypothesis model
+    for default_name, default_value in zip(
+        ["apoptosis_rate", "eta", "name"], [0.001, 1.0e-10, f"{mode}_hypothesis"]
     ):
-        segmented_objects = segmentation_to_objects(segmentation.data[:100, ...])
-        data, properties, graph = run_tracker(segmented_objects, config_file_path)
+        hypothesis_model_dict[default_name] = default_value
+
+    # add hypotheses to hypothesis model
+    hypothesis_model_dict["hypotheses"] = hypotheses
+    motion_model = MotionModel(**motion_model_dict)
+    hypothesis_model = HypothesisModel(**hypothesis_model_dict)
+
+    # add parameters outside the internal models
+    max_search_radius = getattr(container, "max_search_radius").value
+    return TrackerConfig(
+        max_search_radius=max_search_radius,
+        motion_model=motion_model,
+        hypothesis_model=hypothesis_model,
+    )
+
+
+def _tracker_config_to_widgets(container: Container, config: TrackerConfig):
+    """Helper function to update a container's widgets
+    with the values in a given tracker config.
+    """
+    getattr(container, "max_search_radius").value = config.max_search_radius
+    for model in ["motion_model", "hypothesis_model", "object_model"]:
+        model_config = getattr(config, model)
+        if model_config:
+            for parameter, value in model_config:
+                if parameter in hidden_variable_names:
+                    continue
+                if parameter in Matrices().names:
+                    sigma = Matrices.get_sigma(parameter, value)
+                    getattr(container, f"{parameter}_sigma").value = sigma
+                if parameter == "hypotheses":
+                    for hypothesis in all_hypotheses:
+                        getattr(container, hypothesis).value = False
+                    for hypothesis in value:
+                        getattr(container, hypothesis).value = True
+                else:
+                    getattr(container, parameter).value = value
+    mode_is_cell = config.motion_model.A[0, 3] == 1
```

Comment:
What are `0` and `3` here? I suspect row and column but would be good to avoid magic variables